compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingInterfaces.td

```diff
@@ -72,12 +72,61 @@ def IREEEncoding_EncodingLayoutAttrInterface :
         assert(false && "unimplemented interface method");
         return {};
       }]
+    >
+  ];
+}
+
+def IREEEncoding_SerializableEncodingAttrInterface :
+  AttrInterface<"SerializableEncodingAttrInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Encoding";
+  let description = [{
+    Interface for encoding attributes that can be attached on a tensor type. An
+    encoding has two states, serialized and unserialized. An encoding can be set
+    with high level encoded information, e.g., operation type, element types of
+    operands, etc. During the lowering, the virtual encodings are progressively
+    lowered to physical encodings/operations. At some point, an encoding could
+    in the middle state, where the serialized data is encoded in the attribute.
```

Comment:
What do you mean by 'serialized data'?

Reply:
Maybe call it 'finalized data' or 'final serialized data' then?

Reply:
Thanks for the suggestion! I think `final serialized data` is better, because I don't want to introduce new terminology (i.e., `finalized`).